provers/celestia-prover/prover/src/main.rs

```diff
@@ -176,18 +230,65 @@ impl Prover for ProverService {
             .await
             .map_err(|e| Status::internal(e.to_string()))?;
 
-        let proof = self.membership_prover.generate_proof(
+        path_value_and_proofs.iter().for_each(|(path, value, _)| {
+            println!("path: {:?}", path);
+            println!("value: {:?}", value);
+        });
+
+        println!(
+            "Generating proof with path_value_and_proofs: {:?}",
+            path_value_and_proofs
+        );
+
+        let kv_proofs = path_value_and_proofs
+            .into_iter()
+            .map(|(path, value, proof)| {
+                let path = path.into_iter().map(Bytes::from).collect::<Vec<Bytes>>();
+                let value = Bytes::from(value);
+                (KVPair { path, value }, proof)
+            })
+            .collect();
+
+        // Generate the SP1 proof
+        let start_time = Instant::now();
+        let sp1_proof = self.membership_prover.generate_proof(
             trusted_block.signed_header.header.app_hash.as_bytes(),
-            key_proofs,
+            kv_proofs,
+        );
+        let elapsed = start_time.elapsed();
+
+        println!(
+            "Generated proof for height: {:?} in {:.2?}",
+            trusted_block.signed_header.header.height.value() as i64,
+            elapsed
         );
 
+        let membership_output =
+            MembershipOutput::abi_decode(&sp1_proof.public_values.to_vec(), true).unwrap();
+        membership_output.kvPairs.iter().for_each(|kv| {
+            println!(
+                "membership_output path: {:?} value: {:?}",
+                kv.path, kv.value
+            );
+        });
+
+        let membership_proof = MembershipProof::from(SP1MembershipProof {
+            sp1Proof: SP1Proof::new(
+                &self.membership_prover.vkey.bytes32(),
+                sp1_proof.bytes(),
+                sp1_proof.public_values.to_vec(),
+            ),
+            trustedConsensusState: trusted_consensus_state,
+        });
```

Comment:
Wow did you get the zk membership proofs to work! nice job!

Reply:
Yes. TBH it was way easier to use the SP1 programs written in solidity-ibc-eureka than to debug the issues in our mock SP1 programs.